rlog/state-separation.mdx

```diff
@@ -0,0 +1,1255 @@
+
+3. **Execution Processes and UTXO Management**: This section will focus on the lifecycle of UTXOs within the architecture, from their generation to consumption. 
+We will also cover the processes involved in managing UTXOs, including proof generation, state transitions, and ensuring transaction validity.
+
+These topics will provide a comprehensive understanding of how NSSA enables flexible and secure interactions within decentralized applications.
+
+## 1. Architecture's components
+---
+
+The Nescience State Separation Architecture (NSSA) introduces an advanced prototype execution framework designed to enhance privacy and security in blockchain applications. 
+This framework integrates several essential components: the public state, private state, Zero-Knowledge Virtual Machine (zkVM), various execution types, Nescience users, and smart contracts.
+
+### a) Public State
+---
+
+The public state in the Nescience State Separation Architecture (NSSA) is a fundamental component designed to hold all publicly accessible information within 
+the blockchain network. This state is organized as a single Merkle tree structure, a sophisticated data structure that ensures efficient and secure data verification. 
+The public state includes critical information such as user balances and the public storage data of smart contracts.
+
+In an account-based model, the public state operates by storing each account or smart contract's public data as individual leaf nodes within the Merkle tree. 
+When transactions occur, they directly modify the state by updating these leaf nodes. This direct modification ensures that the most current state of the network 
+is always reflected accurately.
+
+The Merkle tree structure is essential for maintaining data integrity. Each leaf node contains a hash of a data block, and each non-leaf node contains the 
+hash of its child nodes. This hierarchical arrangement means that any change in the data will result in a change in the corresponding hash, making it easy to detect 
+any tampering. The root hash, or Merkle root, is stored on the blockchain, providing a cryptographic guarantee of the data's integrity. This root hash serves as a single, 
+concise representation of the entire state, enabling quick and reliable verification by any network participant.
+
+Transparency is a key feature of the public state. All data stored within this state is openly accessible and verifiable by any participant in the network. 
+This openness ensures that all transactions and state changes are visible and auditable, fostering trust and accountability. For example, user balances are
+publicly viewable, which helps ensure transparency and trust in the system. Similarly, public smart contract storage can be accessed and verified by anyone, 
+making it suitable for applications that require public scrutiny and auditability, such as public record updates and financial transactions.
```

Comment:
I wonder if we can elaborate on "financial transactions" or be more specific on a type of financial transaction - there are many financial transactions people would want to keep private.